docker.go

```diff
@@ -46,9 +46,48 @@ type Parameters struct {
 
 // Volume is the docker volume details associated to the container
 type Volume struct {
-	ContainerPath string `json:"containerPath,omitempty"`
-	HostPath      string `json:"hostPath,omitempty"`
-	Mode          string `json:"mode,omitempty"`
+	ContainerPath string            `json:"containerPath,omitempty"`
+	HostPath      string            `json:"hostPath,omitempty"`
+	Mode          string            `json:"mode,omitempty"`
+	Persistent    *PersistentVolume `json:"persistent,omitempty"`
+}
+
+// PeristentVolume declares a Volume to be Persistent, and sets
+// the size (in MiBs) and optional constraints for the Volume.
+type PersistentVolume struct {
+	Type        *string     `json:"type,omitempty"`
+	Size        int         `json:"size,omitempty"`
+	Constraints *[][]string `json:"constraints,omitempty"`
+}
```

Comment:
Do I see correctly that neither `type` nor `constraints` are [documented in the REST API](https://mesosphere.github.io/marathon/docs/generated/api.html#v2_apps_post)? The relevant JSON blob looks like this:

```json
"persistent":{
  "additionalProperties":false,
  "properties":{
     "size":{
        "type":"integer",
        "description":"The size of the persistent volume in MiB",
        "minimum":0
     }
  },
  "type":"object"
}
```

I can only see a definition in the [documentation on local persistent volumes](https://mesosphere.github.io/marathon/docs/persistent-volumes.html).

(Not that it strictly matters for this PR, although I'd have preferred a formal definition. I'd file an upstream issue though.)